Scrapper-Service/utility/adapative_request.py

```diff
@@ -1,20 +1,15 @@
 import requests
 import math
+import backoff
 
 class AdaptiveRequest:
     def __init__(self):
         self.max_wait_time = 10
-        self.num_fail = 0
-        self.num_success = 0
+
+    @backoff.on_exception(
+        backoff.expo, #exponential backoff
+        (requests.HTTPError , requests.ConnectionError), #retry if errors encountered
+        max_time=300 #give up after 300 seconds time
```

Comment:
300 seconds is give up time. Retries happen in exponential manner with increasing wait times. For example this will be a series like
```
2
4
8
16
```
and so on. This series can go to infinity and cause problems. So we use give up time to stop this.
```
if (2+4+8+...>=300)
   stop
```

I may have not understood the problem statement correctly. If that's the case, can you explain it more clearly?

Reply:
* Ok , got it. In that case it should be fine. I got the wrong impression due to how the api looked. You can resolve conflict with current implementation, will merge it.